web/html/src/manager/virtualization/nets/network-properties.js

```diff
@@ -46,8 +47,68 @@ type Props = {
 
 function clearFields(initialModel, setModel) {
   if (initialModel) {
-    // Flatten the model for the form
-    let flattened = flattenModel(initialModel);
+    let definition = initialModel;
+
+    // Convert some int properties to string since validators only work with strings
+    const intToString = (value: Object, key: string) => {
+      if (!_isNil((value || {})[key]) && typeof value[key] === "number") {
+        value[key] = value[key].toString();
+      }
+    };
+
+    (definition.srv || []).forEach(srv => {
+      intToString(srv, "port");
+      intToString(srv, "priority");
+      intToString(srv, "weight");
+    });
+
+    (definition.vlans || []).forEach(vlan => intToString(vlan, "tag"));
+
+    // rename the ipv[46] to ipv[46]def
+    ["ipv4", "ipv6"].forEach(prop => {
+      if (definition[prop] != null) {
+        intToString(definition[prop], "prefix");
+        definition[`${prop}def`] = definition[prop];
+        delete definition[prop];
+      }
+    });
+
+    // Compute the selection fields from the model
+    definition['ipv6-enabled'] = !_isNil(definition.ipv6def);
+    if (!_isNil(definition.pf)) {
+      definition['interface-selection-type'] = 'pf';
+    } else if ((Array.isArray(definition.interfaces) && definition.interfaces.length > 0) || !_isNil(definition.vf)) {
+      definition['interface-selection-type'] = 'vf';
+    }
+    if ((definition.virtualport || {}).type === "802.1qbh") {
+      definition['virtualport-params-type'] = _isNil(definition.virtualport.profileid) ? 'vsi': 'profileid';
+    }
+
+    // Merge the DNS host names into coma-separated string
+    if (!_isNil((definition.dns || {}).hosts)) {
+      definition.dns.hosts = definition.dns.hosts.map(host => ({
+        address: host.address,
+        names: Array.isArray(host.names) ? host.names.join() : host.names,
+      }))
+    }
+
+    // Since the flattenModel would flatten these arrays while they should stay as arrays for the Select component
+    // merge the vf and interfaces into coma-separated string and split them after the flattening
```

Comment:
To me this sounds like `flattenModel()` needs to be reworked instead. If we have to write numerous hacks in components to use a utility, it doesn't look like it serves us well anymore. Iirc there were issues stemming from similar stuff in the previous PR as well?

Reply:
Created a card for this: https://github.com/uyuni-project/uyuni/issues/3392